simtools/applications/derive_mirror_rnda.py

```diff
@@ -82,12 +83,12 @@
         default model is read from the simulation model database.
     mirror_list (file, optional)
         Table with mirror ID and panel radius.
-    use_random_flen (activation mode, optional)
-        Use random focal lengths, instead of the measured ones. The argument random_flen can be
-        used to replace the default random_focal_length from the model.
-    random_flen (float, optional)
+    use_random_focal_length (activation mode, optional)
+        Use random focal lengths, instead of the measured ones. The argument random_focal_length
+        can be used to replace the default random_focal_length from the model.
+    random_focal_length (float, optional)
         Value of the random focal lengths to replace the default random_focal_length. Only used if
-         use_random_flen is activated.
+        'use_random_focal_length' is activated.
     no_tuning (activation mode, optional)
         Turn off the tuning - A single case will be simulated and plotted.
     test (activation mode, optional)
```

Comment:
Does the example in the docstring still runs for you?

I am getting: `ValueError: Invalid model version None in DB CTAO-Simulation-Model-v0-4-0 (allowed are ['6.0.0', '5.0.0'])
`
and if specifying additionally the model version:
`FileNotFoundError: [Errno 2] No such file or directory: 'MLTdata-preproduction.ecsv'`

Truncated output:
`INFO::db_handler(l145)::_find_latest_simulation_model_db::Updated the DB simulation model to the latest version CTAO-Simulation-Model-v0-4-0
Traceback (most recent call last):
  File "/workdir/env/bin/simtools-derive-mirror-rnda", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/workdir/external/simtools/simtools/applications/derive_mirror_rnda.py", line 219, in main
    panel_psf = MirrorPanelPSF(label, args_dict, db_config)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/workdir/external/simtools/simtools/ray_tracing/mirror_panel_psf.py", line 47, in __init__
    self._get_psf_containment()
  File "/workdir/external/simtools/simtools/ray_tracing/mirror_panel_psf.py", line 104, in _get_psf_containment
    _psf_list = Table.read(
                ^^^^^^^^^^^
 `

Reply:
Fixed the docstring example.

As a side note: I don't like the docstrings examples at all. I see the value for the documentation, but at this point there are not tested automatically and they are a duplication of the examples given in the integration test. Ideal would be to sync them somehow.

Reply:
Thanks it runs now (but seing a Warning and the filename looks odd)! I wonder how it generated the filename in such a way. I would expect it should be None from looking at the ModelDataWriter if no output filename is supplied?
Yes agree it would be ideal if it generates the docstring examples from the configuration files.


```
Measured D80:
Mean = 1.400 cm

Simulated D80:
Mean = 1.400 cm, StdDev = 0.017 cm

mirror_random_reflection_angle
Previous value = 0.003000
New value = 0.003905

WARNING::metadata_collector(l120)::get_data_model_schema_file_name::No schema file found.
INFO::model_data_writer(l143)::write::Writing data to /workdir/external/simtools/simtools-output/d-2024-11-01/371423d8-5f1b-4c64-998a-2439c3c043a4-derive_mirror_rnda.ecsv
```

Reply:
You mean "371423d8-5f1b-4c64-998a-2439c3c043a4-derive_mirror_rnda.ecsv
"? This is the default when no metadata is given: a newly generated UUID plus the tool name (something which is in simtools since a long time). Important to note is that this is intermediate output not used for anything. We need to implement the writing of the model parameter (simtools-style), which requires first merging PR #1215.  

Also ignore the warning from the metadata collector (should have mentioned this above) - this will be fixed in the next step (this PR is about testing).